File: frontend/src/shared/branding.ts
/**
 * 이 화면이 무슨 플랫폼인가 — **서버가 말해 준다.**
 *
 * 번들 하나로 여러 플랫폼(허브 · 그룹 쌍둥이들)을 띄운다. 그래서 이름은 빌드에 굽지 않고,
 * 서버가 `index.html` 에 심는 `<meta name="app-name">` 들에서 읽는다(backend/app/main.py —
 * 설치의 `.env` 가 정한다). 첫 화면부터 값이 있으므로 깜빡임이 없다.
 *
 * meta 가 없는 곳은 둘뿐이다 — Vite 개발 서버(vite.config.ts 가 `backend/.env` 에서 같은 meta 를
 * 심는다)와 시험. 그때만 아래 기본값(틀의 이름)이 쓰인다. **기본값을 제품 이름으로 고치지
 * 않는다** — 그러면 모든 설치가 그 이름으로 보인다. 설치의 이름은 `.env` 에.
 *
 * 서버 응답에도 이름이 실려 오므로(`/api/health`) 둘이 어긋나면 서버 화면에서 드러난다.
 */

function meta(name: string): string | null {
  if (typeof document === 'undefined') return null
  return document.querySelector(`meta[name="${name}"]`)?.getAttribute('content') ?? null
}

/** 틀의 기본값 — meta 가 없을 때(개발 서버 · 시험)만. 백엔드 `branding.py` 와 같은 값. */
export const DEFAULT_APP_NAME = 'StandardPlatform'
export const DEFAULT_APP_SLUG = 'standardplatform'
export const DEFAULT_APP_TAGLINE = '사내 플랫폼 공통 틀'

/** 브라우저 탭·로그인 화면·사이드바 머리글. */
export const APP_NAME = meta('app-name') || DEFAULT_APP_NAME

/**
 * 기계가 읽는 이름 — 서버의 `app_slug`. 저쪽에서는 DB 이름·쿠키 이름·토큰 표식을 만들고,
 * 이쪽에서는 localStorage 키만 만든다 — 같은 서버의 다른 경로에 두 플랫폼을 얹는 구성에서
 * 저장소가 출처로 안 갈리기 때문이다.
 */
export const APP_SLUG = meta('app-slug') || DEFAULT_APP_SLUG

/** 한 줄 설명. 로그인 화면과 사이드바가 같은 말을 한다. */
export const APP_TAGLINE = meta('app-tagline') || DEFAULT_APP_TAGLINE

/**
 * 이 설치가 켠 확장(`.env` 의 EXTENSIONS). 화면은 이 목록에 있는 것의 메뉴 · 페이지만 붙인다
 * (`src/extensions/index.ts`). 서버가 같은 목록으로 라우터를 붙이므로 둘은 늘 같다.
 */
export const ENABLED_EXTENSIONS: readonly string[] = (meta('app-extensions') ?? '')
  .split(',')
  .map((s) => s.trim())
  .filter(Boolean)

/** localStorage 키의 앞머리. */
export const STORAGE_PREFIX = APP_SLUG

/** 클라이언트가 만드는 오류의 코드 앞머리. 서버의 ERROR_PREFIX 와 같아야 한다. */
export const ERROR_PREFIX = 'APP'

File: backend/app/branding.py
"""이 코드가 무슨 제품군인가 — **설치가 무슨 플랫폼인가는 여기가 아니라 `.env` 가 정한다.**

이 저장소 하나로 여러 플랫폼(인스턴스)을 띄운다 — 허브 · 그룹 쌍둥이들. 번들은 하나고,
설치마다 `.env` 의 `APP_SLUG` · `APP_NAME` · `APP_TAGLINE` · `EXTENSIONS` 가
다르다(`app/config.py`).
그래서 여기 있는 이름은 **아무것도 안 준 설치의 기본값**(개발 PC · 시험)일 뿐이다. 코드가
제품 이름을 쓰려면 `get_settings().app_name` 을 읽는다 — 여기서 import 하지 않는다.

## 오류 코드 접두사

`APP-<MODULE>-<NNNN>`. 이것만은 **빌드에 박힌다** — 코드 곳곳의 `errors.code()` 가 조립하는
문자열이라 설치마다 다를 이유가 없고, 로그 검색이 한 값으로 걸려야 한다. 포크(다른 제품군)할 때
`ERROR_PREFIX` 를 바꾸고 `app/` 아래를 찾아 바꾼다 —
시험(`tests/architecture/test_boundaries.py`)이
**섞였는지 검사한다.**
"""

from __future__ import annotations

#: `.env` 에 `APP_NAME` 이 없을 때 — 화면 제목 · API 문서 제목 · 기동 로그.
DEFAULT_APP_NAME = "StandardPlatform"

#: `.env` 에 `APP_SLUG` 가 없을 때 — 기계가 읽는 이름. **DB·쿠키·토큰 표식이 전부 여기서
#: 나온다.**
#:
#: 소문자와 숫자만, 한 덩어리로 적는다(`matnexus` · `testscope` · `crossaxtf`).
#: 옆 플랫폼들이 이미 그 규약이고, DB 이름을 눈으로 대조하는 자리가 있어서
#: 한 줄이 튀면 그때마다 「이게 맞나」 를 다시 확인하게 된다.
#:
#: **이 값을 안 바꾸면 옆 플랫폼과 부딪힌다.** 그리고 그 셋은 부딪히는 방식이
#: 전부 조용하다:
#:
#:   DB       공통 틀에서 나온 표는 이름이 같다(users·workspaces·notifications).
#:            같은 DB 를 보면 **오류 없이 남의 계정 표를 읽는다.**
#:   쿠키     쿠키는 포트를 구분하지 않는다. 한쪽 로그인이 다른 쪽 세션을 덮어
#:            **번갈아 로그아웃**되고, 그 원인은 코드 어디에도 없다.
#:   토큰     옆 플랫폼 토큰을 붙여 넣으면 「형식은 맞는데 인증이 안 되는」 상태가
#:            되는데, 그것은 오타와 구별되지 않는다.
#:
#: 그래서 셋을 따로 적지 않고 `Settings` 가 slug 하나에서 만든다 — 따로 적으면 언젠가 하나가
#: 안 바뀌고, 안 바뀐 하나는 위 셋 중 하나로 나타난다.
DEFAULT_APP_SLUG = "standardplatform"

#: `.env` 에 `APP_TAGLINE` 이 없을 때 — 한 줄 설명. 로그인 화면과 사이드바가 같은 말을 한다.
DEFAULT_APP_TAGLINE = "사내 플랫폼 공통 틀"

#: 오류 코드 접두사. **한 저장소에 하나뿐이어야 한다.**
ERROR_PREFIX = "APP"
